src/models/userStorage.ts

```diff
@@ -12,17 +13,48 @@ export class UserStorage {
     let user = UserStorage.users.find((u) => u.userId === profile.id);
     const { id, displayName, mail, givenName, surname } = profile._json;
 
-    // console.log("Microsoft Profile Data:", profile);
-
     if (user) {
       done(null, user);
     } else {
+      const picture = await getProfilePicture(accessToken);
       user = new User(id, displayName, mail)
         .withName(givenName, surname)
         .withAdminStatus(false)
-        .withProvider("Microsoft");
+        .withProvider("Microsoft")
+        .withPicture(picture);
       UserStorage.users.push(user);
       done(null, user);
     }
   }
 }
+
+export async function getProfilePicture(accessToken: string): Promise<string> {
+  console.log(`The access token: ${accessToken}`);
```

Comment:
Sometimes it looks like the `accessToken` it not set:

> profiles-server     | Error fetching profile picture: Error: Failed to fetch profile picture: 401
profiles-server     |     at getProfilePicture (file:///home/node/app/dist/models/userStorage.js:33:19)
profiles-server     |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
profiles-server     |     at async file:///home/node/app/dist/routes/mainRoutes.js:16:13

So maybe add a check for that one?